pkg/codeconfig/codeconfig.go

```diff
@@ -43,48 +44,55 @@ import (
 // CodeConfig - represents a collection of related functions and their shared dependencies.
 type CodeConfig interface {
 	Collect() error
-	Handlers() []string
 	ToStack() (*stack.Stack, error)
 }
 
 type codeConfig struct {
 	// A stack can be composed of one or more applications
-	functions map[string]*FunctionDependencies
-	stackPath string
-	files     []string
-	lock      sync.RWMutex
+	functions    map[string]*FunctionDependencies
+	initialStack *stack.Stack
+	lock         sync.RWMutex
 }
 
-func New(stackPath string, globString string) (CodeConfig, error) {
-	files, err := filepath.Glob(filepath.Join(stackPath, globString))
+func New(s *stack.Stack) (CodeConfig, error) {
+	return &codeConfig{
+		initialStack: s,
+		functions:    map[string]*FunctionDependencies{},
+		lock:         sync.RWMutex{},
+	}, nil
+}
+
+func Populate(initial *stack.Stack) (*stack.Stack, error) {
+	cc, err := New(initial)
 	if err != nil {
 		return nil, err
 	}
 
-	return &codeConfig{
-		stackPath: stackPath,
-		files:     files,
-		functions: map[string]*FunctionDependencies{},
-		lock:      sync.RWMutex{},
-	}, nil
-}
+	err = build.CreateBaseDev(initial)
+	if err != nil {
+		return nil, err
+	}
 
-func (c *codeConfig) Handlers() []string {
-	return c.files
+	err = cc.Collect()
+	if err != nil {
+		return nil, err
+	}
+
+	return cc.ToStack()
 }
 
 // Collect - Collects information about all functions for a nitric stack
 func (c *codeConfig) Collect() error {
 	wg := sync.WaitGroup{}
 	errList := utils.NewErrorList()
 
-	for _, f := range c.files {
+	for _, f := range c.initialStack.Functions {
 		wg.Add(1)
 
 		// run files in parallel
 		go func(file string) {
 			defer wg.Done()
-			rel, err := filepath.Rel(c.stackPath, file)
+			rel, err := filepath.Rel(c.initialStack.Dir, file)
```

Comment:
I think the original glob method we were using was fully qualifying the file paths, this is erroring on attempting to make an unrelated path relative (as it already is relative.)